web/packages/teleport/src/Discover/Shared/ConfigureDiscoveryService/CreatedDiscoveryConfigDialog.tsx

```diff
@@ -0,0 +1,120 @@
+/**
+ * Teleport
+ * Copyright (C) 2023  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React from 'react';
+import {
+  Text,
+  Flex,
+  AnimatedProgressBar,
+  ButtonPrimary,
+  ButtonSecondary,
+  Mark,
+} from 'design';
+import * as Icons from 'design/Icon';
+import Dialog, { DialogContent } from 'design/DialogConfirmation';
+
+import type { Attempt } from 'shared/hooks/useAttemptNext';
+
+export type CreatedDiscoveryConfigDialog = {
+  attempt: Attempt;
+  retry(): void;
+  close(): void;
+  next(): void;
+  region: string;
+  /**
+   * show notification that resources might take some
+   * time to appear after setup has finished.
+   */
+  notifyAboutDelay: boolean;
+};
+
+export function CreatedDiscoveryConfigDialog({
+  attempt,
+  retry,
+  close,
+  next,
+  region,
+  notifyAboutDelay,
+}: CreatedDiscoveryConfigDialog) {
+  let content: JSX.Element;
+  if (attempt.status === 'failed') {
+    content = (
+      <>
+        <Flex mb={5} alignItems="center">
+          <Icons.Warning size="large" ml={1} mr={2} color="error.main" />
+          <Text>{attempt.statusText}</Text>
+        </Flex>
+        <Flex>
+          <ButtonPrimary mr={3} width="50%" onClick={retry}>
```

Comment:
`width="50%"` doesn't seem to do anything here, because the parent flex needs to be set to `width="100%"`. I also prefer to use `flex: 1` instead of `width: x%` when using flexbox, but that doesn't matter much here.

Also, again it doesn't matter much in this particular scenario, but it's a good habit to set `gap={3}` on `Flex` instead of setting a margin on an individual button. With `gap`, there will be no unnecessary space added if elements end up in separate rows instead of next to each other, due to e.g., width of the browser window changing.